jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -252,67 +253,99 @@ public synchronized void closeTracer() {
 
   /**
    * @param statsFactory the factory
-   * @deprecated Use {@link #setStatsFactory(StatsFactory)} instead
+   * @deprecated Use {@link #withStatsFactory(StatsFactory)} instead
    */
   @Deprecated
   public void setStatsFactor(StatsFactory statsFactory) {
     this.statsFactory = statsFactory;
   }
 
+  /**
+   * @deprecated Use {@link #withStatsFactory(StatsFactory)} instead
+   */
+  @Deprecated
   public void setStatsFactory(StatsFactory statsFactory) {
     this.statsFactory = statsFactory;
   }
 
+  public Configuration withStatsFactory(StatsFactory statsFactory) {
```

Comment:
an overlap with #349, I suggest we don't touch setStatsFactory in this PR and let it be deprecated in #349

Reply:
reerted